smartsim/entity/dbnode.py

```diff
@@ -179,119 +195,36 @@ def _parse_db_hosts(self):
-                        if "IPADDRESS:" in content:
-                            ip = content[-1]
-                            ips.append(ip)
-
-            # suppress error
-            except FileNotFoundError:
-                pass
-
-            logger.debug("Waiting for RedisIP files to populate...")
-            if len(ips) < len(self._shard_ids):
-                # Larger sleep time, as this seems to be needed for
-                # multihost setups
-                if len(ips) > 0:
-                    # if we found at least one "IPADDRESS", we know
-                    # output streams all go to the same file
-                    multiple_files = False
-                    break
-                else:
-                    ips = []
-                    time.sleep(5)
-                    continue
-            else:
-                ips = list(dict.fromkeys(ips))
-                return ips
-
-        if multiple_files is None:
-            logger.error("RedisIP address lookup strategy failed.")
-            raise SmartSimError("Failed to obtain database hostname")
-
-        if multiple_files == True:
-            for shard_id in self._shard_ids:
-                trials = 5
-                ip = None
-                filepath = osp.join(self.path, self.name + f"_{shard_id}.out")
-                # try a few times to give the database files time to
-                # populate on busy systems.
-                while not ip and trials > 0:
-                    try:
-                        with open(filepath, "r") as f:
-                            lines = f.readlines()
-                            for line in lines:
-                                content = line.split()
-                                if "IPADDRESS:" in content:
-                                    ip = content[-1]
-                                    break
-
-                    # suppress error
-                    except FileNotFoundError:
-                        pass
-
-                    logger.debug("Waiting for RedisIP files to populate...")
-                    if not ip:
-                        # Larger sleep time, as this seems to be needed for
-                        # multihost setups
-                        time.sleep(5)
-                        trials -= 1
-
-                if not ip:
-                    logger.error("RedisIP address lookup strategy failed.")
-                    raise SmartSimError("Failed to obtain database hostname")
-
-                ips.append(ip)
-
+        # Find out if all shards' output streams are piped to separate files
+        if len(self._output_files) > 1:
+            for output_file in self._output_files:
+                filepath = osp.join(self.path, output_file)
+                ip = self._parse_db_host(filepath)
         else:
-            filepath = osp.join(self.path, self.name + ".out")
-            trials = 5
+            filepath = osp.join(self.path, self._output_files[0])
```

Comment:
I know that it is assumed that `len(self._output_files) >= 1`, but it still may be worth it to make an explicit check to protect against future misuse of the class. Perhaps somewhere during its construction?